### ISSUE TITLE
🟢  feat: Make it possible to name a strategy

### ISSUE THREAD

**Issue by tlevesque-ueat** [Unleash/unleash#3254]
### Describe the feature request

Currently, feature toggle strategies have a `name` property, but it's actually the type of strategy, not a user-provided name. It would be useful to let the user specify a name for each strategy. This would make it easier to identify a strategy when there are multiple strategies with different constraints.

**Comment by thomasheartman** [comment]
Hi, @tlevesque-ueat! 👋🏼 That's a good observation, and I agree with you that it calling it a `type` instead of `name` might be more correct. For semver-compatibility reasons, we wouldn't be able to change that until v5.

However, regarding the feature request: that may be possible to support. I think the idea is good. Let me take it back to the team and figure out what want to do.  Would you be willing to help us develop that feature if it's something we want to look at?

**Comment by tlevesque-ueat** [comment]
> I agree with you that it calling it a `type` instead of `name` might be more correct. For semver-compatibility reasons, we wouldn't be able to change that until v5.

Oh, I wasn't suggesting you should change it, that would be a major breaking change. But maybe just introduce a new `displayName` field?

> Would you be willing to help us develop that feature if it's something we want to look at?

I suppose I could give it a try, but no promises... I don't have a lot of spare time.

**Comment by thomasheartman** [comment]
> Oh, I wasn't suggesting you should change it, that would be a major breaking change. But maybe just introduce a new `displayName` field?

Yeah, I think that's a sensible solution. And yeah, didn't think you were suggesting we change it, but just wanted to add that extra bit of information for completeness ☺️ 

> I suppose I could give it a try, but no promises... I don't have a lot of spare time.

No worries! I'll bring it up internally too, but can't guarantee that we'll be able to prioritize it any time soon. Anyway, let me see what we think internally and then get back to you, yeah?

**Comment by tlevesque-ueat** [comment]
> Anyway, let me see what we think internally and then get back to you, yeah?

Sounds good to me!

**Comment by thomasheartman** [comment]
Okay, so I'm back again! This is something we've heard about from other users too, and it's something we'd like to look into working on at some point. However, it's too early to say whether we'll include it on our list of things to do in Q2 or not. 

So yeah: if it's something you'd like to work on to get it in sooner, we'd be happy to help you land that PR 😄 If you don't think you'll find the time, then that's also perfectly fine, and we'll work on it when we get around to it.

**Comment by tlevesque-ueat** [comment]
Hi @thomasheartman,

Thanks for your reply. I could give it a try, but I would probably need some guidance, since I'm not familiar with the code. I had a quick look, but couldn't immediately figure out where to make the changes.

**Comment by chriswk** [comment]
Hi @tlevesque-ueat - Happy to chat on Slack if you think that helps you. 

For completeness, I'll post to the issue as well.

I would think there are a couple of steps to do here, 

The features_strategies table would need a new column to represent our new displayName (I'd probably call it display_name, we've tended towards keeping snake_case in database tables). Migrations are done using the db-migrate library (have a look at https://docs.getunleash.io/contributing/backend/overview#making-a-schema-change)

Then we'd need to change the endpoints which updates/creates FeatureStrategies to accept this extra parameter. Have a look at `addFeatureStrategy` in `src/lib/routes/admin-api/project/features.ts`

Then we (you?) would need to update the frontend component which displays strategies on a feature to display and edit/add a displayname. I would start by having a look in the `./frontend/src/component/FeatureStrategy/FeatureStrategyForm`

**Comment by tlevesque-ueat** [comment]
@chriswk thanks for the pointers!
I'll get back to you after I made some progress or if I need more help (probably not in the next few days though).

**Comment by tlevesque-ueat** [comment]
I'm unable to even run the application. I'm getting this:

```
[ERROR] KnexTimeoutError: Knex: Timeout acquiring a connection. The pool is probably full. Are you missing a .transacting(trx) call?
    at Function.get (/mnt/c/Projects/unleash/node_modules/knex/lib/knex-builder/make-knex.js:43:26)
    at new KnexStore (/mnt/c/Projects/unleash/node_modules/connect-session-knex/lib/index.js:107:28)
    at sessionDb (/mnt/c/Projects/unleash/dist/lib/middleware/session-db.js:15:17)
    at createApp (/mnt/c/Projects/unleash/dist/lib/server-impl.js:72:53)
    at start (/mnt/c/Projects/unleash/dist/lib/server-impl.js:151:27)
    at processTicksAndRejections (node:internal/process/task_queues:96:5)
    at async /mnt/c/Projects/unleash/dist/server-dev.js:9:9
```

I'm on Windows, if that matters (running in WSL)

**Comment by ivarconr** [comment]
will be available from v5.2
